McEngine/src/GUI/CBaseUIElement.h

```diff
@@ -1,112 +1,154 @@
+	virtual void setAnchorX(float xAnchor) {if (m_vAnchor.x != xAnchor){m_vmPos.x -= m_vmSize.x * (xAnchor - m_vAnchor.x); m_vPos.x -= m_vSize.x * (xAnchor - m_vAnchor.x); m_vAnchor.x = xAnchor; if (m_parent != nullptr) updateLayout(); onMoved();}}
+	virtual void setAnchorY(float yAnchor) {if (m_vAnchor.y != yAnchor){m_vmPos.y -= m_vmSize.y * (yAnchor - m_vAnchor.y); m_vPos.y -= m_vSize.y * (yAnchor - m_vAnchor.y); m_vAnchor.y = yAnchor; if (m_parent != nullptr) updateLayout(); onMoved();}}
+	virtual void setAnchor(Vector2 anchor) {if (m_vAnchor != anchor){m_vmPos -= m_vmSize * (anchor - m_vAnchor); m_vPos -= m_vSize * (anchor - m_vAnchor); m_vAnchor = anchor; if (m_parent != nullptr) updateLayout(); onMoved();}}
+
+	virtual void setAnchorAbsolute(float xAnchor, float yAnchor) {if (m_vAnchor.x != xAnchor || m_vAnchor.y != yAnchor){m_vAnchor.x = xAnchor, m_vAnchor.y = yAnchor;}}
+	virtual void setAnchorAbsoluteX(float xAnchor) {if (m_vAnchor.x != xAnchor) {m_vAnchor.x = xAnchor;}}
+	virtual void setAnchorAbsoluteY(float yAnchor) {if (m_vAnchor.y != yAnchor) {m_vAnchor.y = yAnchor;}}
+	virtual void setAnchorAbsolute(Vector2 anchor) {if (m_vAnchor != anchor) {m_vAnchor = anchor;}}
+
+	virtual void setVisible(bool visible) {m_bVisible = visible;}
+	virtual void setActive(bool active) {m_bActive = active;}
+	virtual void setKeepActive(bool keepActive) {m_bKeepActive = keepActive;}
+	virtual void setDrawManually(bool drawManually) {m_bDrawManually = drawManually;}
+	virtual void setPositionManually(bool positionManually) {m_bPositionManually = positionManually;}
+	virtual void setEnabled(bool enabled) {if (enabled != m_bEnabled) {m_bEnabled = enabled; if (m_bEnabled) {onEnabled();} else {onDisabled();}}}
+	virtual void setBusy(bool busy) {m_bBusy = busy;}
+	virtual void setName(UString name) {m_sName = name;}
+	virtual void setParent(CBaseUIElement *parent) {m_parent = parent;}
+
+	// getters
+	inline const Vector2& getPos() const {return m_vPos;}
+	inline const Vector2& getSize() const {return m_vSize;}
+	inline UString getName() const {return m_sName;}
+	inline const Vector2& getRelPos() const {return m_vmPos;}
+	inline const Vector2& getRelSize() const {return m_vmSize;}
+	inline const Vector2& getAnchor() const {return m_vAnchor;}
+	inline CBaseUIElement *getParent() const {return m_parent;}
+
+	virtual bool isActive() {return m_bActive || isBusy();}
+	virtual bool isVisible() {return m_bVisible;}
+	virtual bool isEnabled() {return m_bEnabled;}
+	virtual bool isBusy() {return m_bBusy && isVisible();}
+	virtual bool isDrawnManually() {return m_bDrawManually;}
+	virtual bool isPositionedManually() {return m_bPositionManually;}
+	virtual bool isMouseInside() {return m_bMouseInside && isVisible();}
+
+	// actions
+	void stealFocus() {m_bMouseInsideCheck = true; m_bActive = false; onFocusStolen();}
+	virtual void updateLayout() {if(m_parent != nullptr) m_parent->updateLayout();}

```

Comment:
Space after 'if'
{if (m_parent != nullptr) m_parent->updateLayout();}

Also, virtual functions should always come before non-virtual functions. In this case, the declaration for updateLayout() should come before stealFocus(). (I'm still trying to have a consistent style for that myself, nitpicking intensifies)